src/hardware/uni-t-ut8802e/protocol.c

```diff
@@ -0,0 +1,430 @@
+/*
+ * This file is part of the libsigrok project.
+ *
+ * Copyright (C) 2023 Konrad Tagowski <konrad.tagowski@grinn-global.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <config.h>
+#include "math.h"
+
+#include "protocol.h"
+
+/*
+ * Developer notes on the UT8802E protocol:
+ * - Serial communication over HID CP2110 converter USB-UART.
+ * - UART frame format 8n1 at 9600 bps.
+ * - DMM packet starts with a magic marker, functionality, value, etc.
+ * - UT8802E only sends a packet with measurement. Example of frame:
+ *   Frame | func | value    | comma | settings | checksum |
+ *   ac    | 1b   | 45 01 00 | 33	 | 04  	    | 44       |
+ * - The frame to send a command to the multimeter is unknown.
+*/
+
+static const int8_t range_volt[]  = {-3,  0};
+static const int8_t range_amp[]   = {-6, -3,  0};
```

Comment:
range_amp[2] is never used. The initializer list can be shortened.

Reply:
Range_amp[2], I used to choose a range for the current measurement in 168 to 194 lines.

Reply:
No, `range_amp[2]` is not used.